### PR TITLE
bytewords: add fuzz target

### PR DESCRIPTION
**Pull request by jeandudey**
Adds a fuzz target for byte words decode function.

Found a crash when the string doesn't have only ASCII characters and also in the minimal decoding added verification for the length so that we don't also crash there too.

### CODE REVIEW

**Comment by dspicher** [comment]
Very nice, thanks for the PR! Can you fix the [Rust build stage](https://github.com/dspicher/ur-rs/actions/runs/3742803444/jobs/6354200178)?

**Comment by jeandudey** [comment]
@dspicher fixed the formatting issue

**Comment by jeandudey** [comment]
I think I'll remove https://github.com/dspicher/ur-rs/pull/172/commits/a31a7b2e3f23d0b58aa9c8d0e450f7560c1f5afc as I'm trying an approach that doesn't use `alloc` to decode `bytewords`. I'll also add some tests for the ASCII and length cases.

**Comment by jeandudey** [comment]
@dspicher updated the PR

File: src/bytewords.rs [hunk 1]
```diff
@@ -228,6 +242,17 @@ mod tests {
...
         assert_eq!(decode("", Style::Standard).unwrap_err(), Error::InvalidWord);
+
+        // invalid length
+        assert_eq!(
+            decode("aea", Style::Minimal).unwrap_err(),
+            Error::InvalidLength
+        );
+
+        // non ASCII
+        assert_eq!(decode("₿", Style::Standard).unwrap_err(), Error::NonAscii);
```
Comment (dspicher) [inline src/bytewords.rs:253]: Hehe nice!

**Review by dspicher — APPROVED** [review]
Thank you!

**Pull request merged**